torchbenchmark/util/model.py

```diff
@@ -483,8 +483,18 @@ def enable_amp(self):
             self.amp_context = lambda: torch.cpu.amp.autocast()
         elif self.device == "cuda":
             self.amp_context = lambda: torch.cuda.amp.autocast()
-        if is_staged_train_test(self):
-            self.forward_contexts.append(self.amp_context)
+        if self.test == "eval":
+            self.add_context(self.amp_context)
+        elif self.test == "train":
+            if is_staged_train_test(self):
+                self.add_context(self.amp_context, TEST_STAGE.FORWARD)
+            else:
+                warnings.warn(
+                        "Usually models only want to enable AMP in forward path, so expected "
+                        "model to have staged train support."
+                        )
+
```

Comment:
In this case, we can keep this warning and add `self.amp_context` to `TEST_STAGE.ALL`. Can you help address this comment?

Reply:
try to added it, and also updated in extra_args.py in amp_fp16 path.

Reply:
I don't think the `amp_fp16` path added in `extra_args.py` is correct, because it is supposed to add `torch.cuda.amp.autocast(dtype=torch.float16)` context, but here we are adding the `torch.cuda.amp.autocast()` context. What do you think?

Reply:
you are right. fixed. then should we also modify the amp_bf16 path when the model not support staged train?

Reply:
We can keep that for now. Later we may want to refactor it to clean up the code.